tests/conftest.py

```diff
@@ -164,7 +165,7 @@ async def redirect_301(scope, receive, send):
     await send({"type": "http.response.body"})
 
 
-SERVER_SCOPE = "session"
+SERVER_SCOPE: Literal["session"] = "session"
```

Comment:
What's prompting this change?
Is it strictly necessary for us to type this variable, here?
Do we have a build failure without it?
Can it simply be `str`?

Reply:
It's precisely because it identifies it as `str` that we need it. This is the mypy issue:

```bash
+ mypy httpx tests
tests/conftest.py:170: error: No overload variant of "fixture" matches argument type "str"
tests/conftest.py:170: note: Possible overload variant:
tests/conftest.py:170: note:     def fixture(fixture_function: None = ..., *, scope: Union[Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']], Callable[[str, Config], Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']]]] = ..., params: Optional[Iterable[object]] = ..., autouse: bool = ..., ids: Union[Iterable[Union[None, str, float, int, bool]], Callable[[Any], Optional[object]], None] = ..., name: Optional[str] = ...) -> FixtureFunctionMarker
tests/conftest.py:170: note:     <1 more non-matching overload not shown>
tests/conftest.py:175: error: No overload variant of "fixture" matches argument type "str"
tests/conftest.py:175: note: Possible overload variant:
tests/conftest.py:175: note:     def fixture(fixture_function: None = ..., *, scope: Union[Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']], Callable[[str, Config], Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']]]] = ..., params: Optional[Iterable[object]] = ..., autouse: bool = ..., ids: Union[Iterable[Union[None, str, float, int, bool]], Callable[[Any], Optional[object]], None] = ..., name: Optional[str] = ...) -> FixtureFunctionMarker
tests/conftest.py:175: note:     <1 more non-matching overload not shown>
tests/conftest.py:181: error: No overload variant of "fixture" matches argument type "str"
tests/conftest.py:181: note: Possible overload variant:
tests/conftest.py:181: note:     def fixture(fixture_function: None = ..., *, scope: Union[Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']], Callable[[str, Config], Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']]]] = ..., params: Optional[Iterable[object]] = ..., autouse: bool = ..., ids: Union[Iterable[Union[None, str, float, int, bool]], Callable[[Any], Optional[object]], None] = ..., name: Optional[str] = ...) -> FixtureFunctionMarker
tests/conftest.py:181: note:     <1 more non-matching overload not shown>
tests/conftest.py:186: error: No overload variant of "fixture" matches argument type "str"
tests/conftest.py:186: note: Possible overload variant:
tests/conftest.py:186: note:     def fixture(fixture_function: None = ..., *, scope: Union[Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']], Callable[[str, Config], Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']]]] = ..., params: Optional[Iterable[object]] = ..., autouse: bool = ..., ids: Union[Iterable[Union[None, str, float, int, bool]], Callable[[Any], Optional[object]], None] = ..., name: Optional[str] = ...) -> FixtureFunctionMarker
tests/conftest.py:186: note:     <1 more non-matching overload not shown>
tests/conftest.py:192: error: No overload variant of "fixture" matches argument type "str"
tests/conftest.py:192: note: Possible overload variant:
tests/conftest.py:192: note:     def fixture(fixture_function: None = ..., *, scope: Union[Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']], Callable[[str, Config], Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']]]] = ..., params: Optional[Iterable[object]] = ..., autouse: bool = ..., ids: Union[Iterable[Union[None, str, float, int, bool]], Callable[[Any], Optional[object]], None] = ..., name: Optional[str] = ...) -> FixtureFunctionMarker
tests/conftest.py:192: note:     <1 more non-matching overload not shown>
tests/conftest.py:198: error: No overload variant of "fixture" matches argument type "str"
tests/conftest.py:198: note: Possible overload variant:
tests/conftest.py:198: note:     def fixture(fixture_function: None = ..., *, scope: Union[Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']], Callable[[str, Config], Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']]]] = ..., params: Optional[Iterable[object]] = ..., autouse: bool = ..., ids: Union[Iterable[Union[None, str, float, int, bool]], Callable[[Any], Optional[object]], None] = ..., name: Optional[str] = ...) -> FixtureFunctionMarker
tests/conftest.py:198: note:     <1 more non-matching overload not shown>
tests/conftest.py:274: error: No overload variant of "fixture" matches argument type "str"
tests/conftest.py:274: note: Possible overload variant:
tests/conftest.py:274: note:     def fixture(fixture_function: None = ..., *, scope: Union[Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']], Callable[[str, Config], Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']]]] = ..., params: Optional[Iterable[object]] = ..., autouse: bool = ..., ids: Union[Iterable[Union[None, str, float, int, bool]], Callable[[Any], Optional[object]], None] = ..., name: Optional[str] = ...) -> FixtureFunctionMarker
tests/conftest.py:274: note:     <1 more non-matching overload not shown>
tests/conftest.py:281: error: No overload variant of "fixture" matches argument type "str"
tests/conftest.py:281: note: Possible overload variant:
tests/conftest.py:281: note:     def fixture(fixture_function: None = ..., *, scope: Union[Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']], Callable[[str, Config], Union[Literal['session'], Literal['package'], Literal['module'], Literal['class'], Literal['function']]]] = ..., params: Optional[Iterable[object]] = ..., autouse: bool = ..., ids: Union[Iterable[Union[None, str, float, int, bool]], Callable[[Any], Optional[object]], None] = ..., name: Optional[str] = ...) -> FixtureFunctionMarker
tests/conftest.py:281: note:     <1 more non-matching overload not shown>
Found 8 errors in 1 file (checked 57 source files)
```
Do you have an alternative suggestion?

Reply:
> Do you have an alternative suggestion?

Nope. Just making sure I understand e'rything going in, and we've got a super clear history in place throughout. 😀